File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/controller/RpcPostmanSceneTestRunnerController.java
package org.javamaster.invocationlab.admin.controller;

import org.javamaster.invocationlab.admin.dto.AbstractCaseDto;
import org.javamaster.invocationlab.admin.dto.SceneCaseDto;
import org.javamaster.invocationlab.admin.dto.UserCaseDto;
import org.javamaster.invocationlab.admin.dto.WebApiRspDto;
import org.javamaster.invocationlab.admin.service.repository.redis.RedisRepository;
import org.javamaster.invocationlab.admin.service.scenetest.SceneTester;
import org.javamaster.invocationlab.admin.util.JsonUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * 批量处理用例
 * 场景测试运行相关的操作
 *
 * @author yudong
 */
@Controller
@RequestMapping("/dubbo-postman/")
public class RpcPostmanSceneTestRunnerController {
    @Autowired
    RedisRepository cacheService;
    @Autowired
    SceneTester requestService;

    @ResponseBody
    @RequestMapping(value = "case/scene/run", method = RequestMethod.POST)
    public WebApiRspDto<Map<String, Object>> runSceneCase(@RequestBody SceneCaseDto sceneDto) {
        List<UserCaseDto> testCaseDtoList = new ArrayList<>(1);
        for (AbstractCaseDto dto : sceneDto.getCaseDtoList()) {
            String jsonStr = cacheService.mapGet(dto.getGroupName(), dto.getCaseName());
            UserCaseDto caseDto = JsonUtils.parseObject(jsonStr, UserCaseDto.class);
            testCaseDtoList.add(caseDto);
        }
        Map<String, Object> rst = requestService.process(testCaseDtoList, sceneDto.getSceneScript());
        return WebApiRspDto.success(rst);
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/controller/RpcPostmanSceneTestRunnerController.kt
package org.javamaster.invocationlab.admin.controller

import org.javamaster.invocationlab.admin.model.dto.SceneCaseDto
import org.javamaster.invocationlab.admin.model.dto.UserCaseDto
import org.javamaster.invocationlab.admin.model.dto.WebApiRspDto
import org.javamaster.invocationlab.admin.service.repository.redis.RedisRepository
import org.javamaster.invocationlab.admin.service.scenetest.SceneTester
import org.javamaster.invocationlab.admin.util.JsonUtils.parseObject
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.bind.annotation.ResponseBody

/**
 * 批量处理用例
 * 场景测试运行相关的操作
 *
 * @author yudong
 */
@Controller
@RequestMapping("/dubbo-postman/")
class RpcPostmanSceneTestRunnerController {
    @Autowired
    private lateinit var cacheService: RedisRepository

    @Autowired
    private lateinit var requestService: SceneTester

    @ResponseBody
    @RequestMapping(value = ["case/scene/run"], method = [RequestMethod.POST])
    fun runSceneCase(@RequestBody sceneDto: SceneCaseDto): WebApiRspDto<Map<String, Any>> {
        val testCaseDtoList: MutableList<UserCaseDto> = ArrayList(1)
        for (dto in sceneDto.caseDtoList) {
            val jsonStr = cacheService.mapGet<String>(dto.groupName!!, dto.caseName!!)
            val caseDto = parseObject(jsonStr, UserCaseDto::class.java)
            testCaseDtoList.add(caseDto)
        }
        val rst = requestService.process(testCaseDtoList, sceneDto.sceneScript!!)
        return WebApiRspDto.success(rst)
    }
}
